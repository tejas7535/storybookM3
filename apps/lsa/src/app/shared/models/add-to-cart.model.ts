export interface AddToCartEventPayload {
  cart: CartItem[];
}

export interface CartItem {
  productCode: string;
  packCode?: string;
  quantity: number;
}
