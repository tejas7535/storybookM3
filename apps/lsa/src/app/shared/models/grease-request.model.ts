export interface GreaseRequest {
  id: string;
  name: string;
}
