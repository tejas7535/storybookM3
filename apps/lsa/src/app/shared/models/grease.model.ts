export interface Grease {
  title: string;
  id: string;
}
