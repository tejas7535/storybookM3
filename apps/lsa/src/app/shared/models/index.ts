export * from './add-to-cart.model';
export * from './grease.model';
export * from './page.model';
export * from './recommendation-form.model';
export * from './recommendation-form-value.model';
export * from './recommendation-request.model';
export * from './recommendation-response.model';
export * from './recommendation-table-data.model';
