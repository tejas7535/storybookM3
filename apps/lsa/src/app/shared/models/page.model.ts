import { Pages } from '../constants/pages.enum';

export interface Page {
  name: Pages;
  completed: boolean;
}
