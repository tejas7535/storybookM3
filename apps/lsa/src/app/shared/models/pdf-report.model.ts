import { ProductGroup } from '@schaeffler/pdf-generator';

export interface LubricatorTableData {
  [key: string]: string;
}

export type ProductDetails = ProductGroup;
