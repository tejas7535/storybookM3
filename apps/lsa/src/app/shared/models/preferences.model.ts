export enum Unitset {
  SI = 'UNITSET_SI',
  FPS = 'UNITSET_FPS',
}
