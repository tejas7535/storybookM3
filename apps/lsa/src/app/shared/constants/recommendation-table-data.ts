import { Lubricator } from '../models';

export type LubricatorType = 'minimum' | 'recommended';

export type RecommendationTableRowConfiguration =
  | {
      type: 'technical';
      fieldName: keyof Lubricator['technicalAttributes'];
    }
  | {
      type: 'composite';
      fieldName: string;
      formatFunction: (device: Lubricator, which: LubricatorType) => string;
    }
  | keyof Lubricator;

type RecommendationTableConfiguration = RecommendationTableRowConfiguration[];

export const recommendationTableConfiguration: RecommendationTableConfiguration =
  [
    'maxOperatingPressure',
    'volume',
    {
      type: 'composite',
      fieldName: 'tempRange',
      formatFunction: (lub: Lubricator) => `${lub.minTemp} - ${lub.maxTemp}Â°C`,
    },
    {
      type: 'technical',
      fieldName: 'mounting_position',
    },
    {
      type: 'technical',
      fieldName: 'func_principle',
    },
  ];
