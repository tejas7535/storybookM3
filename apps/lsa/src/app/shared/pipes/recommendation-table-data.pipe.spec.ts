import { mockResponse } from '@lsa/testing/mocks/recommendation.mock';

import { Lubricator } from '../models';
import { RecommendationTableDataPipe } from './recommendation-table-data.pipe';

jest.mock('@jsverse/transloco', () => ({
  translate: jest.fn((key) => key),
}));

jest.mock('@lsa/shared/constants', () => ({
  ...jest.requireActual('@lsa/shared/constants'),
  recommendationTableFields: ['volume'],
}));

describe('RecommendationTableDataPipe', () => {
  let pipe: RecommendationTableDataPipe;

  beforeEach(() => {
    pipe = new RecommendationTableDataPipe();
  });

  it('should create', () => {
    expect(pipe).toBeDefined();
  });

  describe('transform', () => {
    beforeEach(() => {
      pipe['getFieldValue'] = jest.fn();
    });

    it('should create the data object', () => {
      const result = pipe.transform(mockResponse.lubricators);

      expect(result).toMatchSnapshot();
      expect(pipe['getFieldValue']).toHaveBeenCalledWith(
        'volume',
        mockResponse.lubricators.minimumRequiredLubricator
      );
      expect(pipe['getFieldValue']).toHaveBeenCalledWith(
        'volume',
        mockResponse.lubricators.recommendedLubricator
      );
    });

    it('should create the data object with identical lubricators', () => {
      const result = pipe.transform({
        minimumRequiredLubricator:
          mockResponse.lubricators.minimumRequiredLubricator,
        recommendedLubricator:
          mockResponse.lubricators.minimumRequiredLubricator,
      });

      expect(result).toMatchSnapshot();
      expect(pipe['getFieldValue']).toHaveBeenCalledWith(
        'volume',
        mockResponse.lubricators.minimumRequiredLubricator
      );
      expect(pipe['getFieldValue']).toHaveBeenCalledWith(
        'volume',
        mockResponse.lubricators.minimumRequiredLubricator
      );
    });

    it('should create the data object without lubricators', () => {
      const result = pipe.transform({});

      expect(result).toMatchSnapshot();
      expect(pipe['getFieldValue']).toHaveBeenCalledWith('volume', undefined);
      expect(pipe['getFieldValue']).toHaveBeenCalledWith('volume', undefined);
    });
  });

  describe('getFieldValue', () => {
    beforeEach(() => {
      pipe['getTranslation'] = jest.fn((key) => key);
    });
    it('should do nothing if lubricator is not defined', () => {
      const result = pipe['getFieldValue']('accessories');

      expect(result).not.toBeDefined();
    });

    it.each([
      ['outputDiameter', '1 centimeter', 1, 'centimeter'],
      ['maxOperatingPressure', 'â‰¤ 1 bar', 1, 'bar'],
      ['maxTemp', '1 degreeCelsius', 1, 'degreeCelsius'],
      ['minTemp', '1 degreeCelsius', 1, 'degreeCelsius'],
      ['batteryPowered', 'no', 0, 'no'],
      ['isOptime', 'yes', 1, 'yes'],
      ['volume', '1 millilitre', 1, 'millilitre'],
    ])(
      'should return the transformed field value as string for %p',
      (field, expected, value, expectedKey) => {
        const result = pipe['getFieldValue'](
          field as keyof Lubricator,
          {
            [field]: value,
          } as unknown as Lubricator
        );

        expect(result).toEqual(expected);
        expect(pipe['getTranslation']).toHaveBeenCalledWith(expectedKey);
      }
    );

    it('should return the field value as string in default case', () => {
      const result = pipe['getFieldValue']('noOfOutlets', {
        noOfOutlets: 1,
      } as Lubricator);

      expect(result).toEqual('1');
    });
  });

  describe('getTranslation', () => {
    it('should translate the given key with prefix', () => {
      const result = pipe['getTranslation']('test');

      expect(result).toEqual('recommendation.result.test');
    });
  });
});
