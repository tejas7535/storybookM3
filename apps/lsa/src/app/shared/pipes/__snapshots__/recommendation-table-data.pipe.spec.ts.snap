// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecommendationTableDataPipe transform should create the data object 1`] = `
Object {
  "headers": Object {
    "minimum": Object {
      "description": "CONCEPT1, Single-point lubrication, gas-driven, without lubricant",
      "isRecommended": false,
      "matNr": "0951660500000",
      "name": "ARCALUB-C1-60-REFILLABLE",
      "productImageUrl": "https://cdn.schaeffler-ecommerce.com/cdn/00171550_d.png",
    },
    "recommended": Object {
      "description": "CONCEPT2, Single-point lubrication, gas-driven, without lubricant",
      "isRecommended": true,
      "matNr": "0951660500001",
      "name": "ARCALUB-C2-60-REFILLABLE",
      "productImageUrl": "https://cdn.schaeffler-ecommerce.com/cdn/00171550_d.png",
    },
  },
  "rows": Array [
    Object {
      "field": "maxOperatingPressure",
      "minimum": undefined,
      "recommended": undefined,
    },
    Object {
      "field": "volume",
      "minimum": undefined,
      "recommended": undefined,
    },
    Object {
      "field": "tempRange",
      "minimum": "-20 - 55째C",
      "recommended": "-25 - 60째C",
    },
    Object {
      "field": "mounting_position",
      "minimum": undefined,
      "recommended": undefined,
    },
    Object {
      "field": "func_principle",
      "minimum": undefined,
      "recommended": undefined,
    },
  ],
}
`;

exports[`RecommendationTableDataPipe transform should create the data object with identical lubricators 1`] = `
Object {
  "headers": Object {
    "minimum": undefined,
    "recommended": Object {
      "description": "CONCEPT1, Single-point lubrication, gas-driven, without lubricant",
      "isRecommended": true,
      "matNr": "0951660500000",
      "name": "ARCALUB-C1-60-REFILLABLE",
      "productImageUrl": "https://cdn.schaeffler-ecommerce.com/cdn/00171550_d.png",
    },
  },
  "rows": Array [
    Object {
      "field": "maxOperatingPressure",
      "minimum": undefined,
      "recommended": undefined,
    },
    Object {
      "field": "volume",
      "minimum": undefined,
      "recommended": undefined,
    },
    Object {
      "field": "tempRange",
      "minimum": "-20 - 55째C",
      "recommended": "-20 - 55째C",
    },
    Object {
      "field": "mounting_position",
      "minimum": undefined,
      "recommended": undefined,
    },
    Object {
      "field": "func_principle",
      "minimum": undefined,
      "recommended": undefined,
    },
  ],
}
`;

exports[`RecommendationTableDataPipe transform should create the data object without lubricators 1`] = `
Object {
  "headers": Object {
    "minimum": undefined,
    "recommended": undefined,
  },
  "rows": Array [
    Object {
      "field": "maxOperatingPressure",
      "minimum": undefined,
      "recommended": undefined,
    },
    Object {
      "field": "volume",
      "minimum": undefined,
      "recommended": undefined,
    },
    Object {
      "field": "tempRange",
      "minimum": undefined,
      "recommended": undefined,
    },
    Object {
      "field": "mounting_position",
      "minimum": undefined,
      "recommended": undefined,
    },
    Object {
      "field": "func_principle",
      "minimum": undefined,
      "recommended": undefined,
    },
  ],
}
`;
